sonar-plugin-api/src/main/java/org/sonar/api/batch/fs/internal/FileMetadata.java

```diff
@@ -138,14 +150,25 @@ protected void handleIgnoreEoL(char c) {
     @Override
     protected void newLine() {
       sb.append(LINE_FEED);
-      globalMd5Digest.update(sb.toString().getBytes(StandardCharsets.UTF_8));
+      processBuffer();
       sb.setLength(0);
     }
 
     @Override
     protected void eof() {
       if (sb.length() > 0) {
-        globalMd5Digest.update(sb.toString().getBytes(StandardCharsets.UTF_8));
+        processBuffer();
+      }
+    }
+
+    private void processBuffer() {
+      try {
+        if (sb.length() > 0) {
+          ByteBuffer encoded = encoder.encode(CharBuffer.wrap(sb));
+          globalMd5Digest.update(encoded.array(), 0, encoded.limit());
+        }
+      } catch (CharacterCodingException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)